### PR TITLE
Added XCMS plotting tools

### PR DESCRIPTION
**Pull request by hechth**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-metabolomics/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection

See https://github.com/RECETOX/galaxytools/pull/523 for xref

### CODE REVIEW

File: tools/xcms/xcms_plot_eic.xml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+<tool id="xcms_plot_eic" name="xcms plot eic" version="@TOOL_VERSION@+galaxy0" profile="21.09">
+    <description>Plot the extracted ion chromatogram (EIC) from mzML file</description>
+    <macros>
+        <import>macros.xml</import>
```
Comment (lecorguille) [inline tools/xcms/xcms_plot_eic.xml:4]: Should be `macros_xcms_plot.xml`, isn't it?

File: tools/xcms/xcms_plot_eic.xml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    </macros>
+    <expand macro="bio.tools"/>
+    <expand macro="creator"/>
+    <requirements>
```
Comment (lecorguille) [inline tools/xcms/xcms_plot_eic.xml:8]: Add this in the macros

File: tools/xcms/macros_xcms_plot.xml [hunk 3]
```diff
@@ -0,0 +1,37 @@
+<macros>
+    <token name="@TOOL_VERSION@">4.0.0</token>
+    <xml name="creator">
+    <creator>
```
Comment (lecorguille) [inline tools/xcms/macros_xcms_plot.xml:4]: indents

**Review by lecorguille — CHANGES_REQUESTED** [review]

File: tools/xcms/xcms_plot_eic.xml (hunk 1 above)
Comment (hechth) [reply tools/xcms/xcms_plot_eic.xml:4]: yes

**Comment by hechth** [comment]
@lecorguille think it should be fixed now :) Thanks for reviewing!

**Comment by hechth** [comment]
@lecorguille @bgruening ping

File: tools/xcms/xcms_plot_raw.xml [hunk 4]
```diff
@@ -0,0 +1,55 @@
+<tool id="xcms_plot_raw" name = "xcms plot raw" version="@TOOL_VERSION@+galaxy0" profile="21.09">
```
Comment (bgruening) [inline tools/xcms/xcms_plot_raw.xml:1]:
```suggestion
<tool id="xcms_plot_raw" name="xcms plot raw" version="@TOOL_VERSION@+galaxy0" profile="21.09">
```

**Comment by bgruening** [comment]
@hechth tests are failing...

**Comment by hechth** [comment]
It complains about maximum file size but all added files are tiny @lecorguille and @bgruening

**Comment by bgruening** [comment]
LGTM!

File: tools/xcms/xcms_plot_eic.xml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    <outputs>
+        <data name="output_filename" format="png" label="EIC plot at m/z=$mz_value of $input.element_identifier"  />
+    </outputs>
+    <tests>
+        <test>
+            <param name="input" value="xcms_plot_eic_testdata.mzML"/>
+            <param name="mz_value" value="153.06614"/>
+            <param name="tolerance_ppm" value="10"/>
+            <param name="mslevel" value="1"/>
+            <output name="output_filename" file="eic_plot.png"/>
```
Comment (bgruening) [inline tools/xcms/xcms_plot_eic.xml:40]: Keep in mind that with the latest Galaxy we have all those nice asserts that are better than diif or sim_size comparisons: https://docs.galaxyproject.org/en/latest/dev/schema.html#assertions-for-image-data

Comment (hechth) [reply tools/xcms/xcms_plot_eic.xml:40]: Ah thanks - that is good to know. This is our first plotting tool so wasn't familiar with those!

**Comment by bgruening** [comment]
Test are failing here.

**Comment by hechth** [comment]
Test is failing due to file size but that is invalid ... I don't really know what else to do here.

Maybe the CI is outdated? The tests pass locally - should I try opening the same PR against tools-iuc and see what happens?

**Comment by bgruening** [comment]
Can you maybe update the ci?

**Comment by hechth** [comment]
Hi can we please try to move this forward? We'd really like to use the tools and I'd hate to move them again back over to our repo to speed up the process.

**Comment by zargham-ahmad** [comment]
Hi, can you please trigger the CI @lecorguille and @bgruening

**Comment by bgruening** [comment]
done

**Review by bgruening — APPROVED** [review]

**Review by lecorguille — APPROVED** [review]

**Pull request merged**